packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -259,15 +257,36 @@ abstract class ImplicitlyAnimatedWidget extends StatefulWidget {
 ///
 /// This is the type of one of the arguments of [TweenVisitor], the signature
 /// used by [AnimatedWidgetBaseState.forEachTween].
+///
+/// Instances of this function are expected to take a value and return a tween
+/// beginning at that value.
+/// beginning at that value.
 typedef TweenConstructor<T> = Tween<T> Function(T targetValue);
 
-/// Signature for callbacks passed to [AnimatedWidgetBaseState.forEachTween].
+/// Signature for callbacks passed to [ImplicitlyAnimatedWidgetState.forEachTween].
+///
+/// [ImplicitlyAnimatedWidgetState.forEachTween] will be passed an instance of
+/// this visitor and is expected to call the visitor with the appropriate
+/// arguments for each tween that the state manages -- and to set the tween
+/// value to the return value of this visitor.
+///
+/// The `tween` argument should contain the current tween value. This will
+/// initially be null when the state is first initialized.
+///
+/// The `targetValue` argument should contain the value toward which the state
+/// is animating. For instance, if the state was animating its widget's
+/// "opacity" value, then this argument should contain the widget's current
+/// opacity value.
```

Comment:
Maybe unify the quotes around opacity? Either use them in both instances or don't?